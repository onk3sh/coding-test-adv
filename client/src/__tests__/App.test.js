import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

let App;
beforeAll(() => {
  jest.mock('../components/Carousel', () => ({
    __esModule: true,
    default: () => {
      return 'Render Carousel';
    }
  }));

  jest.mock('../components/Categories', () => ({
    __esModule: true,
    default: () => {
      return 'Render Categories';
    }
  }));

  App = require('../App').default;
});

// By right we should be able to ignore this test
// Really it does nothing but just hold a state and call child component
describe('Test App Component', () => {
  it('should render all child component', async () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot(`
    <div class="app"><header class="app-header">Render CategoriesRender Carousel</header></div>
    `);
  });
});
