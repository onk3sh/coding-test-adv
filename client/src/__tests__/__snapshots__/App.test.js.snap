// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component should render all child component: 
    <div class="app"><header class="app-header">Render CategoriesRender Carousel</header></div>
     1`] = `
<div>
  <div
    class="app"
  >
    <header
      class="app-header"
    >
      Render Categories
      Render Carousel
    </header>
  </div>
</div>
`;